### PR TITLE
docs: add linked library story

### PR DESCRIPTION
**Pull request by filipesilva**
Blocked on https://github.com/angular/angular-cli/pull/6276.

/cc @clydin

### CODE REVIEW

File: docs/documentation/stories/linked-library.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Linked libraries
+
+While working on a library, it is common to use [npm link](https://docs.npmjs.com/cli/link) to
+have avoid reinstalling the library on every build.
+
+While this is very useful there are a few caveats to keep in mind.
+
+## The library needs to be AOT compatible
+
+Angular CLI does static analysis even without the `--aot` flag in order to detect lazy-loader routes.
```
Comment (sumitarora) [inline docs/documentation/stories/linked-library.md]: `lazy-loaded`

File: docs/documentation/stories/linked-library.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+If your library is not AOT compatible, you will likely get an static analysis error.
```
Comment (sumitarora) [inline docs/documentation/stories/linked-library.md]: get a static analysis error.

File: docs/documentation/stories/linked-library.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+## The library still needs to be rebuilt on every change
+
+Angular libraries are usually built using TypeScript and thus require to be built before they
+are published.
+For simple cases a linked library might work even without a build step, but this is the exception
```
Comment (sumitarora) [inline docs/documentation/stories/linked-library.md]: cases,

File: docs/documentation/stories/linked-library.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+rather than the norm.
+
+If a library is not being built using it's own build step, then it is being compiled by the
```
Comment (sumitarora) [inline docs/documentation/stories/linked-library.md]: its own build step

File: docs/documentation/stories/linked-library.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+When linking a library remember to have your build step running in watch mode and the library's
+`package.json` pointing at the correct entry points (e.g. 'main' should point at a `.js` file, not
+a `.ts` file).
+
+## Use TypesScript path mapping for Peer Dependencies
+
+Angular libraries should list all `@angular/*` dependencies as
+[Peer Dependencies](https://nodejs.org/en/blog/npm/peer-dependencies/).
+This insures that, when modules asks for Angular, they all get the exact same module.
```
Comment (sumitarora) [inline docs/documentation/stories/linked-library.md]: This ensures that, when modules ask for Angular,

File: docs/documentation/stories/linked-library.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+If a library lists `@angular/core` in `dependencies` instead of `peerDependencies` then it might
+get a *different* Angular module instead, which will cause your application to break.
+
+While developing a library however, you'll need to have all of your peer dependencies also installed
```
Comment (sumitarora) [inline docs/documentation/stories/linked-library.md]: library,

File: docs/documentation/stories/linked-library.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+located in it's `node_modules` folder.
+This can cause problems while building or running your application.
+
+To get around this problem you can use the TypeScript
+[path mapping](https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping).
+With it, you can tell TypeScript that it should load some modules from a specific location.
+
+You should list all the peer dependencies that your library uses in `./tsconfig.json`, pointing
+them at the local copy in the apps `node_modules` folder.
+This ensures that your all will always load the local copies of the modules your library asks for.
```
Comment (sumitarora) [inline docs/documentation/stories/linked-library.md]: you will always

File: docs/documentation/stories/linked-library.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
+# Linked libraries
+
+While working on a library, it is common to use [npm link](https://docs.npmjs.com/cli/link) to
+have avoid reinstalling the library on every build.
```
Comment (clydin) [inline docs/documentation/stories/linked-library.md]: Remove "have"

File: docs/documentation/stories/linked-library.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+```
+{
+  "compilerOptions": {
+    // ...
+    // Note: these paths are relative to `baseUrl` path.
+    "paths": {
+      "@angular/common": [
+        "../node_modules/@angular/common"
+      ],
+      "@angular/compiler": [
```
Comment (clydin) [inline docs/documentation/stories/linked-library.md]: These could probably be simplified to just one:
"@angular/*": [ "../node_modules/@angular/*" ]

Works with tsc but would need to confirm it works with the CLI.

File: docs/documentation/stories/linked-library.md (hunk 1 above)
Comment (filipesilva) [inline docs/documentation/stories/linked-library.md]: Fixed

File: docs/documentation/stories/linked-library.md (hunk 2 above)
Comment (filipesilva) [inline docs/documentation/stories/linked-library.md]: Fixed

File: docs/documentation/stories/linked-library.md (hunk 4 above)
Comment (filipesilva) [inline docs/documentation/stories/linked-library.md]: Fixed

File: docs/documentation/stories/linked-library.md (hunk 5 above)
Comment (filipesilva) [inline docs/documentation/stories/linked-library.md]: Fixed

File: docs/documentation/stories/linked-library.md (hunk 3 above)
Comment (filipesilva) [inline docs/documentation/stories/linked-library.md]: Fixed

File: docs/documentation/stories/linked-library.md (hunk 6 above)
Comment (filipesilva) [inline docs/documentation/stories/linked-library.md]: Fixed

File: docs/documentation/stories/linked-library.md (hunk 7 above)
Comment (filipesilva) [inline docs/documentation/stories/linked-library.md]: Fixed

File: docs/documentation/stories/linked-library.md (hunk 8 above)
Comment (filipesilva) [inline docs/documentation/stories/linked-library.md]: Fixed.

File: docs/documentation/stories/linked-library.md (hunk 9 above)
Comment (filipesilva) [inline docs/documentation/stories/linked-library.md]: Checked it and seems to be fine. Fixed.

**Pull request merged**